### DISCUSSION TITLE
Login Error when using bearer and Cookie...

### DISCUSSION THREAD

**Discussion by ghost** [fastapi-users/fastapi-users#1291] [Q&A]
Hi,

We are  trying to implement login feature using something similar to the below code....
```
curl \
-H "Content-Type: multipart/form-data" \
-X POST \
-F "username=king.arthur@camelot.bt" \
-F "password=guinevere" \
http://localhost:8000/auth/jwt/login
```

But we are getting 422 error when hitting the api...
Here is the error we are getting

```
{
    "detail": [
        {
            "type": "json_invalid",
            "loc": [
                "body",
                0
            ],
            "msg": "JSON decode error",
            "input": {},
            "ctx": {
                "error": "Expecting value"
            }
        }
    ]
}
```

Please let us know what would be the fix for this.

Thanks

**Comment by quillan86** [comment]
I am working with @carthick  on this issue that we need help on - to be more clear, we are using a Bearer transport and Database strategy like so:

```

async def get_user_manager(user_db: SQLModelUserDatabase = Depends(get_user_db)):
    yield UserManager(user_db)

bearer_transport = BearerTransport(tokenUrl="auth/db/login")


def get_database_strategy(
    access_token_db: AccessTokenDatabase[AccessToken] = Depends(get_access_token_db),
) -> DatabaseStrategy:
    return DatabaseStrategy(access_token_db, lifetime_seconds=3600)


auth_backend = AuthenticationBackend(
    name="db",
    transport=bearer_transport,
    get_strategy=get_database_strategy,
)

fastapi_users = FastAPIUsers[User, uuid.UUID](get_user_manager, [auth_backend])
```

with the routers defined as such:

```
router = APIRouter(
)

router.include_router(
    fastapi_users.get_auth_router(auth_backend, requires_verification=True),
    prefix="/auth/db",
    tags=["auth"]
)
router.include_router(
    fastapi_users.get_register_router(UserRead, UserCreate),
    prefix="/auth",
    tags=["auth"],
)
router.include_router(
    fastapi_users.get_reset_password_router(),
    prefix="/auth",
    tags=["auth"],
)
router.include_router(
    fastapi_users.get_verify_router(UserRead),
    prefix="/auth",
    tags=["auth"],
)
router.include_router(
    fastapi_users.get_users_router(UserRead, UserUpdate),
    prefix="/user",
    tags=["user"],
)
```
---
- Python: 3.10.13
- FastAPI: 0.103.0
- FastAPI-Users: 12.1.2
---

We can successfully register and verify users - running into this issue with login, which we know is supposed to be form data.

**Comment by frankie567** [comment]
Have you tried out from the Swagger UI (likely on `http://localhost:8000/docs`)? You should be able to login by clicking on the `Authorize` button and input login/password there.

**Comment by quillan86** [comment]
Just tried it via swagger documentation via the Authorize button - it says `Auth Error Error: response status is 404` - good it's not a validation error there I suppose. The user I tried does exist, is verified, and is a superuser.

**Comment by frankie567** [comment]
Check if you have another router with the prefix `/auth/db` that could mask the login router.

**Comment by quillan86** [comment]
The only other endpoint of /auth/db/ is /auth/db/logout; there's no other routers that have that.

**Comment by ghost** [comment]
@frankie567 Do we need to change anything in the code

**Comment by frankie567** [comment]
I don't see anything wrong with what you showed me. I tried on my side and it works correctly. Without seeing the whole project, it's hard to see where the problem might come from.

Try to disable your own routers, middlewares and other things that might get in the way.

**Comment by quillan86** [comment]
We found the issue (thanks for pointing out middleware) - we had a Sanitation Middleware that was interfering with it. Disabling the middleware allowed it to start working again.
